server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -476,6 +508,11 @@ public Short parse(XContentParser parser, boolean coerce) throws IOException {
                 return parser.shortValue(coerce);
             }
 
+            @Override
+            public BiFunction<Script, ScriptService, MapperScript> compiler(String fieldName) {
+                return LONG.compiler(fieldName);
```

Comment:
Is it ok to "silently" promote here? I think I'd prefer a more conservative "no, you can't make this one calculated, use long" message for now and to relax it in the future.

Reply:
++, have changed these to throw explicit errors